### PR TITLE
Split up BridgeAdapter by chain [TKR-402]

### PR DESCRIPTION
**Pull request by mzhu25**
## Description
Splits the `BridgeAdapter` contract up into one contract per chain, in order to buy us some more time wrt contract size. 

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Comment by dextracker** [comment]
Nice one, looks good to me. quick question, this will make it so we have to load the right bridge adapter by chainId in AS? Or will pointing to the right fqt address be enough?

**Review by dextracker — APPROVED** [review]

**Comment by mzhu25** [comment]
> Nice one, looks good to me. quick question, this will make it so we have to load the right bridge adapter by chainId in AS? Or will pointing to the right fqt address be enough?

No the chainId check is so that we don't accidentally deploy the wrong bridge adapter for a particular chain. As far as AS is concerned, having the right FQT address is enough

File: contracts/zero-ex/contracts/src/transformers/bridges/AvalancheBridgeAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        IERC20TokenV06 sellToken,
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount
+    )
+        public
+        override
+        returns (uint256 boughtAmount)
+    {
+        uint128 protocolId = uint128(uint256(order.source) >> 128);
+        if (protocolId == BridgeProtocols.CURVE) {
```
[the remark this answers is not in the document]
Comment (mzhu25) [reply contracts/zero-ex/contracts/src/transformers/bridges/AvalancheBridgeAdapter.sol:65]: good catch

**Review by dekz — APPROVED** [review]
Nice. Have any thoughts on a nice way to query what it supports? So we can prevent a foot gun where we enable in AS but it's not deployed on chain? 

Could error on startup rather than discover in Simbot/prod

**Pull request merged**